build-tools/automation/azure-pipelines.yaml

```diff
@@ -34,12 +32,18 @@ stages:
   # Check - "Xamarin.Android (Code Analysis CredScan)"
   - job: run_static_analysis
     displayName: CredScan
-    pool: $(HostedWinVS2019)
+    pool: $(XA.Build.Win.Pool)
```

Comment:
I'd prefer to keep as much of the pipeline configuration in YAML as possible - could we restore the `HostedMacMojave` and `HostedMac` variables and their usages? Ideally for new machine pool usage we can just introduce one new variable set by the pipeline, something like `$(OnPremMacMojave)`?

Reply:
I think the existing variable names are falling out of touch with what they mean as we start teasing apart trusted/untrusted builds. To encourage YAML reuse between the trusted and untrusted pipelines, we need this abstraction.

There is no need to conditionally set these variables IMO because there is no circumstance that an untrusted build should run on the untrusted pool (and likewise for the trusted pool). Writing in the condition only increases the chance of having the build get assigned to the wrong pool.

Reply:
I'm thinking more about this -- would it make sense to keep the declarations of the pools in the YAML and then have the XA.Build.*.Pool point to the right variable (E.g. HostedWinVS2019)?

Reply:
That way you can keep the bot pool configuration in YAML, which I think is what you want?

Reply:
> I think the existing variable names are falling out of touch with what they mean as we start teasing apart trusted/untrusted builds. To encourage YAML reuse between the trusted and untrusted pipelines, we need this abstraction.

The existing hosted pool variable definitions in YAML are simply a convenience that allows us to avoid having to type the name of a machine pool more than once. I am not entirely sure how the existing hosted machine pool usage is relevant to the trusted/untrusted conversation. I don't see a need to change anything surrounding the jobs which won't be using the new on-prem pools. Am I missing something there?

My thinking was that we would introduce one new variable for the on-prem machines (named `$(OnPremMacMojave)` or `$(OnPremMacMojavePoolName)`, the value of which would be set (and differ) based on the [variable configuration](https://dev.azure.com/devdiv/DevDiv/_apps/hub/ms.vss-ciworkflow.build-ci-hub?_a=edit-build-definition&id=11410&view=Tab_Variables) of the pipeline that picked up the job. The "mac build" and "notarize + upload installers" jobs could then be updated to use this new variable.

Reply:
Ah, gotcha. I think I was a little more focused on doing more work in the renaming now because we will likely also want to swap over the test hosts to on-prem machines eventually. If you would prefer we limit the change to as few places as possible for now and deal with the rest as it comes up, I can swap the names back.

I did see the XA.Build.*.Pool variables and didn't realize they were there for legacy reasons, so it makes sense to not tinker around with those for now.

Reply:
Yeah, let's wait to introduce pool naming changes for those other jobs in a future PR.